packages/react/src/components/Authenticator/FederatedIdentities/context/DisplayTextContext.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+export function displayText(displayName: string): string {
+  return `Sign in with ${displayName}`;
+}
+
+const DisplayTextContext = React.createContext<typeof displayText | undefined>(
+  undefined
+);
+
+export const DisplayTextProvider = ({
+  children,
+  customDisplayText,
+}: {
+  children?: React.ReactNode;
+  customDisplayText?: typeof displayText;
+}): JSX.Element => {
+  return (
+    <DisplayTextContext.Provider value={customDisplayText}>
+      {children}
+    </DisplayTextContext.Provider>
+  );
+};
+
+export const useDisplayTextContext = (): typeof displayText => {
+  const _displayText = React.useContext(DisplayTextContext);
+
+  if (!_displayText) {
```

Comment:
Nit: would it be more clear to just pass `displayText` to `React.createContext` on line 7 instead of `undefined`? Then we wouldn't need to do this check as displayText from the context would always be defined.